webpack.config.cjs

```diff
@@ -4,6 +4,78 @@ const { TsconfigPathsPlugin } = require( 'tsconfig-paths-webpack-plugin' );
 const FileManagerPlugin = require( 'filemanager-webpack-plugin' );
 const webpack = require( 'webpack' );
 const MiniCssExtractPlugin = require( 'mini-css-extract-plugin' );
+const fs = require( 'fs' );
+
+// @TODO: Sample paths, need to update
+const SCHEMA_SRC_DIR = './schema';
+const SCHEMA_OUTPUT_NAME = 'schema.json';
+const SCHEMA_OUTPUT_PATHS = [ './dist/config', './public/config' ];
+
+// Create a custom plugin to emit the merged JSON file
+class EmitMergedJsonPlugin {
+	apply( compiler ) {
+		compiler.hooks.emit.tapAsync(
+			'EmitMergedJsonPlugin',
+			async ( compilation, callback ) => {
+				try {
+					const mergedContent = JSON.stringify(
+						await mergeJsonFiles( SCHEMA_SRC_DIR ),
+						null,
+						2
+					);
+
+					// Write to all output paths
+					await Promise.all(
+						SCHEMA_OUTPUT_PATHS.map( async ( outputPath ) => {
+							await fs.promises.mkdir( outputPath, {
+								recursive: true,
+							} );
+							await fs.promises.writeFile(
+								path.join( outputPath, SCHEMA_OUTPUT_NAME ),
+								mergedContent
+							);
+						} )
+					);
+
+					// Also emit for webpack output
+					compilation.assets[ SCHEMA_OUTPUT_NAME ] = {
```

Comment:
I'm getting the following warning when building locally, I'm assuming it's related to this?

```
(node:11588) [DEP_WEBPACK_COMPILATION_ASSETS] DeprecationWarning: Compilation.assets will be frozen in future, all modifications are deprecated.
BREAKING CHANGE: No more changes should happen to Compilation.assets after sealing the Compilation.
	Do changes to assets earlier, e. g. in Compilation.hooks.processAssets.
	Make sure to select an appropriate stage from Compilation.PROCESS_ASSETS_STAGE_*.
(Use `node --trace-deprecation ...` to show where the warning was created)
```

Reply:
No worries, glad that it's fixed :)